packages/core/src/Provider/SharedWalletProvider/types.ts

```diff
@@ -0,0 +1,32 @@
+import { Ed25519KeyHashHex } from '@cardano-sdk/crypto';
+
+type ScriptType = number;
+
+export const MULTI_SIG_LABEL = 1854;
+
+interface MultiSigParticipant {
+  name: string;
+  description?: string;
+  icon?: string;
+}
+
+interface MultiSigParticipants {
+  [key: string]: MultiSigParticipant;
+}
+
+interface MultiSigRegistration {
+  types: ScriptType[];
+  name?: string;
+  description?: string;
+  icon?: string;
+  participants?: MultiSigParticipants;
+}
+
+export interface MultiSigTransaction {
+  tx_hash: string;
+  json_metadata: MultiSigRegistration;
```

Comment:
- Please update field names to be consistent with cardano-js-sdk style
- Please use opaque string types where applicable
- Please add native scripts, as it will be required for creating the wallet in `lace`
```suggestion
  txId: Cardano.TransactionId;
  metadata: MultiSigRegistration;
  nativeScripts: Cardano.NativeScript[];
```

Reply:
`nativeScripts` is not returned from `/metadata/txs/labels/{Metadata_num}`. Where would that be gotten from?

Reply:
CIP says
> Auxiliary data with a non-empty native scripts array.

Which means you need tx->auxiliaryData->scripts. I don't see Blockfrost endpoint for auxiliary scripts. If there is none, it means you need to fetch entire tx cbor `https://docs.blockfrost.io/#tag/cardano--transactions/GET/txs/{hash}/cbor``

Reply:
Fixed and squashed with 3492029c1dfb6e4160065e24d3445016ae4000d7